### PR TITLE
Fty srr cmd

### PR DESCRIPTION
**Pull request by stark-dev**
First SRR cli implementation, based on `MessageBus` requests

### CODE REVIEW

File: server/src/fty-srr-cmd.cc [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <vector>
+
+#define END_POINT "ipc://@/malamute"
+#define AGENT_NAME "fty-srr-cmd"
+#define AGENT_NAME_REQUEST_DESTINATION "fty-srr-ui"
+#define MSG_QUEUE_NAME "ETN.Q.IPMCORE.SRR.UI"
+#define DEFAULT_TIME_OUT 300
```
Comment (perrettecl) [inline server/src/fty-srr-cmd.cc:50]: 300 s seems short for me. Better to have 1h

**Review by perrettecl — APPROVED** [review]

**Review by perrettecl — CHANGES_REQUESTED** [review]
Update of timeout, otherwise it's ok for me.

File: server/src/fty-srr-cmd.cc [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+#define END_POINT "ipc://@/malamute"
+#define AGENT_NAME "fty-srr-cmd"
+#define AGENT_NAME_REQUEST_DESTINATION "fty-srr-ui"
+#define MSG_QUEUE_NAME "ETN.Q.IPMCORE.SRR.UI"
+#define DEFAULT_TIME_OUT 3600
+
+using namespace dto::srr;
+
+template <typename T>
+std::ostream &operator<< (std::ostream &os, const std::vector<T> &vec)
```
Comment (etnKlendathu) [inline server/src/fty-srr-cmd.cc:55]: fty::implode

File: server/src/fty-srr-cmd.cc [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+{
+    auto it = vec.begin();
+    for(; it != vec.end() -1; it++) {
+        os << *it << ", ";
+    }
+    os << *it;
+    return os;
+}
+
+dto::UserData sendRequest (const std::string &action,
```
Comment (etnKlendathu) [inline server/src/fty-srr-cmd.cc:65]: how long string len do you have in clang format?

**Review by etnKlendathu — COMMENTED** [review]
just added 2 stupid notes...

File: server/src/fty-srr-cmd.cc (hunk 2 above)
Comment (stark-dev) [reply server/src/fty-srr-cmd.cc:55]: Nice, I'll have a look to that

File: server/src/fty-srr-cmd.cc (hunk 3 above)
Comment (stark-dev) [reply server/src/fty-srr-cmd.cc:65]: Probably it was old template, I will update it

**Pull request merged**